plugins/inputs/nfsstat/nfsstat.go

```diff
@@ -0,0 +1,289 @@
+}
+
+func getNFSVers(s string) (int, bool) {
+	if strings.HasPrefix(s, prefix_nfs3) {
+		return 3, true
+	}
+
+	if strings.HasPrefix(s, prefix_nfs4) {
+		return 4, true
+	}
+
+	return 0, false
+}
+
+func getNFSv3Ops() []string {
+	return []string{
+		"null",
+		"getattr",
+		"setattr",
+		"lookup",
+		"access",
+		"readlink",
+		"read",
+		"write",
+		"create",
+		"mkdir",
+		"symlink",
+		"mknod",
+		"remove",
+		"rmdir",
+		"rename",
+		"link",
+		"readdir",
+		"readdirplus",
+		"fsstat",
+		"fsinfo",
+		"pathconf",
+		"commit",
+	}
+}
+
+func getNFSv4Ops() []string {
+	return []string{
+		"null",
+		"read",
+		"write",
+		"commit",
+		"open",
+		"open_confirm",
+		"open_noattr",
+		"open_downgrade",
+		"close",
+		"setattr",
+		"fsinfo",
+		"renew",
+		"setclientid",
+		"setclientid_confirm",
+		"lock",
+		"lockt",
+		"locku",
+		"access",
+		"getattr",
+		"lookup",
+		"lookup_root",
+		"remove",
+		"rename",
+		"link",
+		"symlink",
+		"create",
+		"pathconf",
+		"statfs",
+		"readlink",
+		"readdir",
+		"server_caps",
+		"delegreturn",
+		"getacl",
+		"setacl",
+		"fs_locations",
+		"release_lockowner",
+		"secinfo",
+		"fsid_present",
+		"exchange_id",
+		"create_session",
+		"destroy_session",
+		"sequence",
+		"get_lease_time",
+		"reclaim_complete",
+		"layoutget",
+		"getdeviceinfo",
+		"layoutcommit",
+		"layoutreturn",
+		"secinfo_no_name",
+		"test_stateid",
+		"free_stateid",
+		"getdevicelist",
+		"bind_conn_to_session",
+		"destroy_clientid",
+		"seek",
+		"allocate",
+		"deallocate",
+		"layoutstats",
+		"clone",
+		"copy",
+		"offload_cancel",
+		"lookupp",
+		"layouterror",
+		"copy_notify",
+		"getxattr",
+		"setxattr",
+		"listxattrs",
+		"removexattr",
+	}
+}
+
+func getNFSOp(index int, nfsvers int) (string, error) {
+
+	var nfsops []string
+	switch nfsvers {
+	case 3:
+		nfsops = getNFSv3Ops()
+	case 4:
+		nfsops = getNFSv4Ops()
+	default:
+		return "", ErrorInvalidNFSVersion
+	}
+
+	// The operation index is defined by the slice with the operation
+	// names. Catch potential programming errors if index is out of bounds
+	if index < 0 || index >= len(nfsops) {
+		return "", ErrorInvalidNFSOpIndex
+	}
+
+	return nfsops[index], nil
+}
+
+func parseCounters(s string) ([]int64, error) {
+	ss := strings.Split(s, " ")
+	sn := make([]int64, len(ss))
+	for i, val := range ss {
+		val_i, err := strconv.ParseInt(val, 0, 64)
+		if err != nil {
+			return []int64{}, err
+		}
+		if val_i < 0 {
+			//This should never happen. Lines come from the kernel
+			// and must not have negative or invalid numbers.
+			return []int64{}, ErrorInvalidCounterValue
+		}
+		sn[i] = val_i
+	}
+	return sn, nil
+}
+
+func parseStatLine(line string, nfsvers int) ([]int64, error) {
+
+	var s string
+
+	switch nfsvers {
+	case 3:
+		s = line[len(prefix_nfs3):]
+	case 4:
+		s = line[len(prefix_nfs4):]
+	default:
+		return []int64{}, ErrorInvalidNFSVersion
+	}
+
+	stats, err := parseCounters(s)
+	if err != nil {
+		return []int64{}, err
```

Comment:
Can return nils instead of empty slices. `len(nil) == 0` anyway.
```suggestion
		return nil, err
```